app/assets/javascripts/application.js

```diff
@@ -55,6 +55,110 @@ var ready = function() {
 		}
 	});
 
+	if ('.panel-sort'.length) {
+
+		// Sort panels by stats
+
+		$('.panel-sort').change(function() {
+
+			// get class name of panels and containers
+			var panelClass, panelContainer;
+			if (this.id == "courses") {
+				panelClass = ".course-panel";
+				panelContainer = "#department-courses";
+			} else {
+				panelClass = ".prof-panel";
+				panelContainer = ".prof-panel-container";
+			}
+
+			// id of stat to sort by		
+			var sortString = ".course-" + $(this).find('.active')[0].id,
+				// class of which panels are displayed (current semester or all)
+				selectorString = $("#all").parent().hasClass("active") ? panelClass.concat(".all") : panelClass.concat("current")
+				// how many panels there are (to know when to trigger the next animation)
+				numPanels = $(selectorString).length,
+				slidPanels = 0,
+				// sorted list of professors
+				panelList = sortPanels($(panelClass), sortString);			
+			// if the number of panels displayed is small enough to have smooth animations, animate the change
+			if (numPanels < 50) {
```

Comment:
I'm always in favor of more performance. This transition was used to be in sync with switching from Semester to All.

The threshold itself is somewhat arbitrary but it exists because that many simultaneous animations is very slow.